akka-contrib/src/main/scala/akka/contrib/metrics/stage/CircuitBreakerEventsStage.scala

```diff
@@ -0,0 +1,73 @@
+package akka.contrib.metrics.stage
+
+import akka.contrib.metrics.Event
+import akka.pattern.CircuitBreaker
+import akka.stream.OverflowStrategies._
+import akka.stream._
+import akka.stream.impl.Buffer
+import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler }
+
+private[metrics] class CircuitBreakerEventsStage(breaker: CircuitBreaker, maxBuffer: Int, overflowStrategy: OverflowStrategy)
+  extends GraphStage[SourceShape[Event]] {
+
+  private val out = Outlet[Event]("CircuitBreakerEventsStage.out")
+
+  override def shape: SourceShape[Event] = SourceShape.of(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with OutHandler {
+    var buffer: Buffer[Event] = _
+
+    override def preStart(): Unit = {
+      if (maxBuffer > 0) buffer = Buffer(maxBuffer, materializer)
+
+      breaker.onOpen(callback.invoke(Event.BreakerOpened(now)))
+      breaker.onHalfOpen(callback.invoke(Event.BreakerHalfOpened(now)))
+      breaker.onClose(callback.invoke(Event.BreakerClosed(now)))
+
+      breaker.onCallSuccess(e ⇒ callback.invoke(Event.CallSuccess(now, e)))
```

Comment:
there will be quite a lot of overhead (generating actor messages via these async callbacks) so I think this call must be sampled by default (configurable)